clients/common/rocblas_gentest.py

```diff
@@ -247,6 +247,24 @@ def setdefaults(test):
             test.setdefault('stride_x', int(test['stride_scale']))
             test.setdefault('stride_y', int(test['stride_scale']))
 
+    if test['function'] in ('trsm_strided_batched', 'trsm_ex_strided_batched'):
+        if all([x in test for x in ('N', 'ldb', 'stride_scale')]):
+            test.setdefault('stride_b', int(test['N'] * test['ldb'] *
+                                            test['stride_scale']))
+        if test['side'].upper() == 'L':
+            if all([x in test for x in ('M', 'lda', 'stride_scale')]):
+                test.setdefault('stride_a', int(test['M'] * test['lda'] *
+                                                test['stride_scale']))
+                test.setdefault('stride_c', int(test['M'] * test['lda'] *
+                                                test['stride_scale']))
+        else:
+            if all([x in test for x in ('N', 'lda', 'stride_scale')]):
+                test.setdefault('stride_a', int(test['N'] * test['lda'] *
+                                                test['stride_scale']))
+                test.setdefault('stride_a', int(test['N'] * test['lda'] *
+                                                test['stride_scale']))
```

Comment:
`stride_a` set twice?

The same expressions used twice? Better (unsure of the logic and the order of the if-nests -- indentation is critical):
```
     if all([x in test for x in ('M', 'lda', 'stride_scale')]):
         ldM = int(test['M'] * test['lda'] * test['stride_scale'])
         test.setdefault('stride_a", ldM)
         test.setdefault('stride_c', ldM)
```
and:
```
         ldN = int(test['N'] * test['lda'] * test['stride_scale'])
         test.setdefault('stride_a', ldN)
         test.setdefault('stride_a', ldN)    # ??? stride_a twice

Reply:
Also, if there is no overlap between functions, they should be exclusive (this is assuming the above indention is correct):

```
    if test['function'] in ('asum_strided_batched', 'nrm2_strided_batched',
                            'scal_strided_batched', 'swap_strided_batched',
                            'copy_strided_batched', 'dot_strided_batched',
                            'dotc_strided_batched', 'rot_strided_batched',
                            'rotm_strided_batched'):
        if all([x in test for x in ('N', 'incx', 'stride_scale')]):
            test.setdefault('stride_x', int(test['N'] * abs(test['incx']) *
                                            test['stride_scale']))
        if all([x in test for x in ('N', 'incy', 'stride_scale')]):
            test.setdefault('stride_y', int(test['N'] * abs(test['incy']) *
                                            test['stride_scale']))
        # we are using stride_c for param in rotm
        if all([x in test for x in ('stride_scale')]):
            test.setdefault('stride_c', int(test['stride_scale']) * 5)

    elif test['function'] in ('gemv_strided_batched', 'ger_strided_batched'):
        if test['function'] in ('ger_strided_batched') or test['transA'] in ('T', 'C'):
            if all([x in test for x in ('M', 'incx', 'stride_scale')]):
                test.setdefault('stride_x', int(test['M'] * abs(test['incx']) *
                                                test['stride_scale']))
            if all([x in test for x in ('N', 'incy', 'stride_scale')]):
                test.setdefault('stride_y', int(test['N'] * abs(test['incy']) *
                                                test['stride_scale']))
        else:
            if all([x in test for x in ('N', 'incx', 'stride_scale')]):
                test.setdefault('stride_x', int(test['N'] * abs(test['incx']) *
                                                test['stride_scale']))
            if all([x in test for x in ('M', 'incy', 'stride_scale')]):
                test.setdefault('stride_y', int(test['M'] * abs(test['incy']) *
                                                test['stride_scale']))

    # we are using stride_c for arg c and stride_d for arg s in rotg
    # these are are single values for each batch
    elif test['function'] in ('rotg_strided_batched'):
        if 'stride_scale' in test:
            test.setdefault('stride_a', int(test['stride_scale']))
            test.setdefault('stride_b', int(test['stride_scale']))
            test.setdefault('stride_c', int(test['stride_scale']))
            test.setdefault('stride_d', int(test['stride_scale']))

    # we are using stride_a for d1, stride_b for d2, and stride_c for param in
    # rotmg. These are are single values for each batch, except param which is
    # a 5 element array
    elif test['function'] in ('rotmg_strided_batched'):
        if 'stride_scale' in test:
            test.setdefault('stride_a', int(test['stride_scale']))
            test.setdefault('stride_b', int(test['stride_scale']))
            test.setdefault('stride_c', int(test['stride_scale']) * 5)
            test.setdefault('stride_x', int(test['stride_scale']))
            test.setdefault('stride_y', int(test['stride_scale']))

    elif test['function'] in ('trsm_strided_batched', 'trsm_ex_strided_batched'):
        if all([x in test for x in ('N', 'ldb', 'stride_scale')]):
            test.setdefault('stride_b', int(test['N'] * test['ldb'] *
                                            test['stride_scale']))
        if test['side'].upper() == 'L':
            if all([x in test for x in ('M', 'lda', 'stride_scale')]):
                test.setdefault('stride_a', int(test['M'] * test['lda'] *
                                                test['stride_scale']))
                test.setdefault('stride_c', int(test['M'] * test['lda'] *
                                                test['stride_scale']))
        else:
            if all([x in test for x in ('N', 'lda', 'stride_scale')]):
                test.setdefault('stride_a', int(test['N'] * test['lda'] *
                                                test['stride_scale']))
                test.setdefault('stride_a', int(test['N'] * test['lda'] *
                                                test['stride_scale']))
```

Reply:
Unsure of the duplicate or why I ever had stride_c in trsm since it isn't used. Made if statements exclusive. Made indentation easier to read as per suggested. See 64aa460.